node/src/components/block_proposer.rs

```diff
@@ -353,20 +355,30 @@ impl BlockProposerReady {
     /// Checks if a deploy is valid (for inclusion into the next block).
     fn is_deploy_valid(
         &self,
-        deploy: &DeployHeader,
+        deploy_type: &DeployType,
         block_timestamp: Timestamp,
         deploy_config: &DeployConfig,
         past_deploys: &HashSet<DeployHash>,
     ) -> bool {
+        if deploy_config.max_payment_cost != Motes::zero()
+            && deploy_type.is_transfer()
+            && deploy_type
+                .payment_amount()
+                .expect("transfer has payment amount")
+                > deploy_config.max_payment_cost
```

Comment:
I have to admit this is the first time I see `max_payment_cost` — but isn't this about how much it costs the system to _execute_ the payment, and not about the maximum amount allowed to transfer in a single transaction?

Reply:
Yes, I think it is the latter. I don't know what it's part of the deploy config though. It should be in the contract runtime. @mpapierski @henrytill @EdHastingsCasperLabs can one of you confirm/deny it?

Reply:
@goral09 We don't currently support variable max payment costs on contract runtime. For now we have a hard coded `MAX_PAYMENT` value which we use https://github.com/CasperLabs/casperlabs-node/blob/41463933e2794e32c84581b6abc29e212fc30510/execution_engine%2Fsrc%2Fcore%2Fengine_state%2Fmod.rs#L98. On contract runtime it means the gas limit of a payment code, and also the minimum balance user is required to posses

Reply:
Thanks, I've removed this check.